### ISSUE TITLE
Question: Training and cross validation scoring, am I doing it right?

### ISSUE THREAD

**Issue by adriangb** [EpistasisLab/tpot#1014]
I have a general workflow where I am testing different classifiers on my data. Initially, my workflow looked like this (using `KNeighborsClassifier` as an example):
```python3
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.neighbors import KNeighborsClassifier


classifier = KNeighborsClassifier()
X = ...
y = ...
X = StandardScaler().fit(X).transform(X.astype(float))
score = np.mean(cross_val_score(estimator=classifier, X=X, y=y, cv=5))
```
I think this part is okay according to the [`sklearn` documentation](https://scikit-learn.org/stable/modules/cross_validation.html).

So I'm trying to incorporate `TPOT` into this scheme. I did the following, which works, but I want to make sure I'm doing it right:
```python3
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import StandardScaler
from tpot import TPOTClassifier


classifier = TPOTClassifier(max_time_mins=120, n_jobs=-1),
X = ...
y = ...
classifier.train(X, y)   # selects the best pipeline using CV internally
# get the best pipeline TPOT found
best_pipeline = classifier.fitted_pipeline_
# score the pipeline (will retrain but keep the model / hyperparameters)
score = np.mean(cross_val_score(estimator=best_pipeline, X=X, y=y, cv=5))
```

One particular question: should I be splitting the data into test/train at any point? My understanding is that ideally, I would split my data into train and test, then use the train set with `TPOTClassifier.fit` to find the ideal pipeline and then apply `cross_val_score` on that pipeline using the test set. Or even take it one step further and use nested cross-validation (as referenced in #554 ), which would look something like `score = np.mean(cross_val_score(estimator=TPOTClassifier(), X=X, y=y, cv=CROSS_VAL_FOLDS))))`. But how would that work for the vanilla `sklearn` classifiers (since there is no model selection, just 1 call to `fit`)? And if doing nested cross-validation, TPOT might select a different pipeline each fold, so how would I know which is truly the best overall?

Thank you

**Comment by weixuanfu** [comment]
TPOT internally use `cross_val_score` to evaluate pipelines, so `score = np.mean(cross_val_score(estimator=best_pipeline, X=X, y=y, cv=5))` should provide same score (if cv is the same and there is no `sample_weight`) after `classifier.fit(X, y)`. I think train/test split is recommended to evaluate the performance of best pipeline via holdout score on test dataset.   

 I think the nested cross-validation should be not necessary and too time-consuming for using TPOT, which is like runing TPOT N-fold times. You could manually perform CV with TPOT but each fold should provide a best pipeline which may be different with others. You can build a ensemble model from all best pipelines (via [VotingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.VotingClassifier.html)) to get the best classifier
